src/error.rs

```diff
@@ -71,5 +71,5 @@ impl From<std::io::Error> for Error {
     }
 }
 
-/// Result type of `tokio-socks`
-pub type Result<T> = std::result::Result<T, Error>;
+///// Result type of `tokio-socks`
+//pub type Result<T> = std::result::Result<T, Error>;
```

Comment:
I prefer to keep the `Result` type. Then, we can omit the `Error` in other code.

Reply:
No problem, think there were a few places where I'd have to use `std::result::Result` but I don't mind either way